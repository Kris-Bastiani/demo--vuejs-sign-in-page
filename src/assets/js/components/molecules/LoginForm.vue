<template lang="pug">
	form(title='login form', @submit='e => signIn(e)')
		input-field(label='Email:', placeholder='Enter email', type='email', required, v-model='email')

		input-field(label='Password:', placeholder='Enter password', type='password', required, v-model='password')

		a.dsp__block.mrg_b__lrg(href='/new-password.pug') Want to reset your password?

		submit-button(label='Login')
</template>

<script>
import postData from '../../helpers/postData';
import InputField from '../atoms/InputField.vue';
import SubmitButton from './SubmitButton.vue';

export default {
	components: {
		InputField,
		SubmitButton,
	},
	data: () => ({
		email: '',
		password: '',
	}),
	methods: {
		signIn(e) {
			e.preventDefault();

			const { email, password } = this;

			postData('https://reqres.in/api/login', { email, password })
				.then(data => console.log('Success:', data))
				.catch(console.error);
		},
	},
};
</script>
