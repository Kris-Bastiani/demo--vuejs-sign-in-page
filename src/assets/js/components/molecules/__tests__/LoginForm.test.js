import 'regenerator-runtime/runtime';
import { shallowMount } from '@vue/test-utils';
import LoginForm from '../LoginForm.vue';

describe('LoginForm', () => {
	let wrapper;

	beforeAll(() => { wrapper = shallowMount(LoginForm); });

	it('Matches snapshot', () => { expect(wrapper.element).toMatchSnapshot(); });
});
