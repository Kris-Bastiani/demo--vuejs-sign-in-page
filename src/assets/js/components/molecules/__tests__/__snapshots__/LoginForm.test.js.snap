// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm Matches snapshot 1`] = `
<div>
  <p
    class="message _success"
    style="display: none;"
  >
    You have signed in successfully.
  </p>
  <form
    title="login form"
  >
    <input-field-stub
      data-cy="email-input"
      label="Email:"
      placeholder="Enter email"
      required="true"
      type="email"
      value=""
    />
    <input-field-stub
      data-cy="password-input"
      label="Password:"
      placeholder="Enter password"
      required="true"
      type="password"
      value=""
    />
    <a
      class="dsp__inline-block mrg_b__lrg"
      data-cy="reset-password-link"
      href="/new-password.pug"
    >
      Want to reset your password?

    </a>
    <submit-button-stub
      class=""
      data-cy="submit-button"
      disabled="true"
      label="Login"
    />
  </form>
</div>
`;
